pkg/cmd/repo/clone/clone.go

```diff
@@ -175,12 +175,19 @@ func cloneRun(opts *CloneOptions) error {
 			upstreamName = canonicalRepo.Parent.RepoOwner()
 		}
 
-		_, err = gitClient.AddRemote(ctx, upstreamName, upstreamURL, []string{canonicalRepo.Parent.DefaultBranchRef.Name}, git.WithRepoDir(cloneDir))
+		gc := gitClient.Copy()
+		gc.RepoDir = cloneDir
+
+		_, err = gc.AddRemote(ctx, upstreamName, upstreamURL, []string{canonicalRepo.Parent.DefaultBranchRef.Name})
```

Comment:
I am not exactly following. The `WithRepoDir` modifier does keep the git client directory unchanged and just modifies the single command, but in this case we were now up to 3 commands that needed to be modified so it seemed like an appropriate time to create a new git client specifically targeted at the `cloneDir` directory.

Reply:
My mistake! I thought it modified the `RepoDir` field on the git client rather than appending a `-C` just for this command.

I actually quite like that pattern even if used multiple times. Generally, I find copy semantics in languages with limited mutation control to be difficult to reason about and a regular source of "oops I modified this thing here and I didn't expect it to be modified there".

Not a worry for this though, I think it's a sensible pattern to explore. Thanks for the correction.